samples/boards/stm32/power_mgmt/blinky/src/main.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/gpio.h>
+
+
+#define SLEEP_TIME_MS   2000
+
+static const struct gpio_dt_spec led =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+
+void main(void)
+{
+	bool led_is_on = true;
+
+	if (!device_is_ready(led.port)) {
+		printk("Device not ready\n");
+		return;
+	}
+
+	while (true) {
+		if (led_is_on == true) {
+			gpio_pin_configure_dt(&led, GPIO_OUTPUT_ACTIVE);
+		}
+		gpio_pin_set(led.port, led.pin, (int)led_is_on);
+		if (led_is_on == false) {
+			/* Release resource to allow sleep mode */
+			gpio_pin_configure(led.port, led.pin, GPIO_DISCONNECTED);
```

Comment:
shouldn't it be `gpio_pin_configure_dt(&led, GPIO_DISCONNECTED)` or `gpio_pin_configure(led.port, led.pin, led.dt_flags | GPIO_DISCONNECTED)` ?

Reply:
As mentioned, I've implemented a quite binary check on driver side to identify if port is in  use or not.
If `flag != GPIO_DISCONNECTED`, I consider it is busy.
So configuring dt flags will prevent the system to enter sleep states.

Reply:
That seems fair, though unless there's some value in having pull and drain flags have an effect when the GPIO is disconnected (which would inevitably be driver-specific, as some hardware wouldn't respect that) the driver could just as well determine that if a GPIO is neither an input nor an output it's not in use, regardless of other flags.  I believe that's consistent with the discussion in #32220.

Reply:
@pabigot Indeed. Aim of this first version was rather to check the dynamic behavior of the implementation vs PM framework.
I'm working in parallel on a finer grain approach.